internal/p2p/downloads.go

```diff
@@ -7,55 +7,72 @@ import (
 )
 
 func ProcessDownload(metadata store.BaseMetadata) (bool, error) {
-	returnValue := false
-
-	var fresult bool
-	if appconfInstance.SecurityCapability > 0 {
-		fmt.Println("[INFO] File Security checking CID: " + metadata.IPFSCID + " with SHA256: " + metadata.FileSHA256)
-		// TODO: 				if appconfInstance.SecurityCapability [1 2 3 4]
-
-		if appconfInstance.SecurityCapability <= 3 {
-			fmt.Println("[INFO] Fetching CID: " + metadata.IPFSCID)
-			// ipfs get
-			psfs.GetFileIPFS(metadata.IPFSCID, appconfInstance.CacheFolder+"/"+metadata.IPFSCID+"."+metadata.FileType)
-
-			result, err := psfs.ClamScanFileClean(appconfInstance.CacheFolder + "/" + metadata.IPFSCID + "." + metadata.FileType)
-			if err != nil {
-				return returnValue, err
-			}
-			fresult = result
-		}
+	if appconfInstance.SecurityCapability == int(SecurityCapabilityNone) {
+		fmt.Println("[ERROR] No security capability configured")
+		return false, nil
+	}
 
-		if appconfInstance.SecurityCapability == 4 {
-			if appconfInstance.FFSkipVT {
-				fresult = true
-			} else {
-				result, err := psfs.GetVirusTotalWSVerdictByHash(metadata.FileSHA256) // true == safe
-				if err != nil {
-					return returnValue, err
-				}
-				// fmt.Println("[INFO] File Security check verdict for CID: " + metadata.IPFSCID + " with SHA256: " + metadata.FileSHA256)
-				fresult = result
-				fmt.Println("[INFO] Fetching CID: " + metadata.IPFSCID)
-				// ipfs get
-				psfs.GetFileIPFS(metadata.IPFSCID, appconfInstance.CacheFolder+"/"+metadata.IPFSCID+"."+metadata.FileType)
-			}
-		}
+	fmt.Println("[INFO] File Security checking CID: " + metadata.IPFSCID + " with SHA256: " + metadata.FileSHA256)
+
+	fresult, err := performSecurityScan(metadata)
+	if err != nil {
+		return false, err
+	}
+
+	if !fresult {
+		fmt.Println("[ERROR] File Security check failed for CID: " + metadata.IPFSCID + " with SHA256: " + metadata.FileSHA256)
+		return false, nil
+	}
+
+	// Validate file type
+	ftype, err := psfs.ValidateFileType(appconfInstance.CacheFolder + "/" + metadata.IPFSCID + "." + metadata.FileType)
+	if err != nil {
+		return false, err
+	}
+
+	fmt.Println("[INFO] File Security type check passed for CID: " + metadata.IPFSCID + "." + metadata.FileType)
+
+	if !ftype {
+		return false, nil
+	}
+
+	psfs.PinFileIPFS(metadata.IPFSCID)
+	fmt.Println("[INFO] IPFS Pinned for CID: " + metadata.IPFSCID)
+	return true, nil
+}
+
+// performSecurityScan handles the security scanning based on the configured capability.
+func performSecurityScan(metadata store.BaseMetadata) (bool, error) {
+	capability := SecurityCapability(appconfInstance.SecurityCapability)
+	cachePath := appconfInstance.CacheFolder + "/" + metadata.IPFSCID + "." + metadata.FileType
```

Comment:
Same feedback as above regarding building OS-agnostic paths and  avoiding concatenation.